realm/realm-library/src/objectServer/java/io/realm/mongodb/RealmAsyncResultTask.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb;
```

Comment:
It would have been ideal if this could have been more generic, but if we don't want to use this class for anything in the "base" API. This is fine. It also makes the callback a bit more clear as we can use the App.Callback` instead of coming up with yet another callback interface.

Reply:
I had the exact same thought. I reached the same conclusion as you regarding the callback, as it seems it's very app-oriented, so I opted to leave it like this. I thought of creating a `Callback` interface and have `App.Callback` implement it, which would have allowed having other types of callbacks as well, but this would have entailed having different callback-handling mechanisms too. I'm not entirely sure I have the knowledge to determine which is the best option though...